cli/command/cli.go

```diff
@@ -384,38 +398,21 @@ func (cli *DockerCli) CurrentContext() string {
 
 // StackOrchestrator resolves which stack orchestrator is in use
 func (cli *DockerCli) StackOrchestrator(flagValue string) (Orchestrator, error) {
-	var ctxOrchestrator string
-
-	configFile := cli.configFile
-	if configFile == nil {
-		configFile = cliconfig.LoadDefaultConfigFile(cli.Err())
-	}
-
 	currentContext := cli.CurrentContext()
-	if currentContext == "" {
-		currentContext = configFile.CurrentContext
+	ctxRaw, err := cli.ContextStore().GetContextMetadata(currentContext)
+	if store.IsErrContextDoesNotExist(err) {
+		// case where the currentContext has been removed (CLI behavior is to fallback to using DOCKER_HOST based resolution)
```

Comment:
Not for this PR, but something similar came up in another discussion with @simonferquel - I think this should be an error; we should not silently fall back to `DOCKER_HOST` (because then I may be running commands against the wrong environment, which is likely not what you want)